app/javascript/mastodon/components/hover_card.tsx

```diff
@@ -0,0 +1,205 @@
+
+      if (target === cardRef.current?.parentNode) {
+        clearTimeout(leaveTimerRef.current);
+      }
+    },
+    [setOpen, setAccountId, setAnchor],
+  );
+
+  const handleAnchorMouseLeave = useCallback(
+    (e: MouseEvent) => {
+      if (e.target === anchor || e.target === cardRef.current?.parentNode) {
+        clearTimeout(leaveTimerRef.current);
+        clearTimeout(enterTimerRef.current);
+
+        leaveTimerRef.current = setTimeout(() => {
+          anchor?.removeAttribute('aria-describedby');
+          setOpen(false);
+          setAnchor(null);
+        }, leaveDelay);
+      }
+    },
+    [setOpen, setAnchor, anchor],
+  );
+
+  const handleFocus = useCallback(
+    (e: FocusEvent) => {
+      const { target } = e;
+
+      if (
+        target instanceof HTMLElement &&
+        target.matches('[data-hover-card]')
+      ) {
+        target.setAttribute('aria-describedby', 'hover-card');
+        setAnchor(target);
+        setOpen(true);
+        setAccountId(target.getAttribute('data-hover-card') ?? undefined);
+      }
+    },
+    [setAnchor, setOpen, setAccountId],
+  );
+
+  const handleBlur = useCallback(
+    (e: FocusEvent) => {
+      const { target } = e;
+
+      if (
+        target instanceof HTMLElement &&
+        target.matches('[data-hover-card]')
+      ) {
+        target.removeAttribute('aria-describedby');
+        setOpen(false);
+        setAnchor(null);
+      }
+    },
+    [setOpen, setAnchor],
+  );
+
+  const handleClose = useCallback(() => {
+    clearTimeout(leaveTimerRef.current);
+    clearTimeout(enterTimerRef.current);
+    setOpen(false);
+    setAnchor(null);
+  }, [setOpen, setAnchor]);
+
+  useEffect(() => {
+    document.body.addEventListener('mouseenter', handleAnchorMouseEnter, {
+      passive: true,
+      capture: true,
+    });
+    document.body.addEventListener('mouseleave', handleAnchorMouseLeave, {
+      passive: true,
+      capture: true,
+    });
+    document.body.addEventListener('focus', handleFocus, {
+      passive: true,
+      capture: true,
+    });
+    document.body.addEventListener('blur', handleBlur, {
+      passive: true,
+      capture: true,
+    });
+
+    return () => {
+      document.body.removeEventListener('mouseenter', handleAnchorMouseEnter);
+      document.body.removeEventListener('mouseleave', handleAnchorMouseLeave);
+      document.body.removeEventListener('focus', handleFocus);
+      document.body.removeEventListener('blur', handleBlur);
+    };
+  }, [handleAnchorMouseEnter, handleAnchorMouseLeave, handleFocus, handleBlur]);
+
+  useEffect(() => {
+    if (accountId) {
+      dispatch(fetchAccount(accountId));
+    }
+  }, [dispatch, accountId]);
+
+  if (!accountId) return null;
+
+  return (
+    <Overlay
+      rootClose
+      onHide={handleClose}
+      show={open}
+      target={anchor}
+      placement='bottom-start'
+      flip
+      offset={offset}
+    >
+      {({ props }) => (
+        <div {...props}>
+          <div
```

Comment:
I would extract the tooltip in its own component (which will fetch the account from the state)